keras_retinanet/models/resnet.py

```diff
@@ -19,66 +19,69 @@
 import keras_resnet.models
 from ..models import retinanet
 
-WEIGHTS_PATH_NO_TOP_50 = 'https://github.com/fizyr/keras-models/releases/download/v0.0.1/ResNet-50-model.keras.h5'
-WEIGHTS_PATH_NO_TOP_101 = 'https://github.com/fizyr/keras-models/releases/download/v0.0.1/ResNet-101-model.keras.h5'
-WEIGHTS_PATH_NO_TOP_152 = 'https://github.com/fizyr/keras-models/releases/download/v0.0.1/ResNet-152-model.keras.h5'
+resnet_filename = 'ResNet-{}-model.keras.h5'
+resnet_resource = 'https://github.com/fizyr/keras-models/releases/download/v0.0.1/{}'.format(resnet_filename)
 
 custom_objects = retinanet.custom_objects.copy()
 custom_objects.update(keras_resnet.custom_objects)
 
 
-def ResNet50RetinaNet(inputs, num_classes, weights='imagenet', *args, **kwargs):
-    image = inputs
+def resnet_retinanet(num_classes, backbone='50', inputs=None, weights='imagenet', **kwargs):
+    allowed_backbones = ['50', '101', '152']
+    if backbone not in allowed_backbones:
+        raise ValueError('Backbone (\'{}\') not in allowed backbones ({}).'.format(backbone, allowed_backbones))
 
-    # load pretrained imagenet weights?
+    # choose default input
+    if inputs is None:
+        inputs = keras.layers.Input(shape=(None, None, 3))
+
+    # determine which weights to load
     if weights == 'imagenet':
+        filename = resnet_filename.format(backbone)
+        resource = resnet_resource.format(backbone)
+        if backbone == '50':
+            checksum = '3e9f4e4f77bbe2c9bec13b53ee1c2319'
+        elif backbone == '101':
+            checksum = '05dc86924389e5b401a9ea0348a3213c'
+        elif backbone == '152':
+            checksum = '6ee11ef2b135592f8031058820bb9e71'
+
         weights_path = keras.applications.imagenet_utils.get_file(
-            'ResNet-50-model.keras.h5',
-            WEIGHTS_PATH_NO_TOP_50, cache_subdir='models', md5_hash='3e9f4e4f77bbe2c9bec13b53ee1c2319'
+            filename,
+            resource,
+            cache_subdir='models',
+            md5_hash=checksum
         )
+    elif weights == 'none':
+        weights_path = None
```

Comment:
I wouldn't allow `'none'` as string. Like `weights='imagenet'` it claims another valid filename as a special case. I'd force the user to pass a real `None` instead.

In my opinion, special handling for strings is something better done in argument parsing of a script than in API functions.

Reply:
I like it :+1: